p2p/client.go

```diff
@@ -16,36 +19,63 @@ import (
 	"github.com/libp2p/go-libp2p-peer"
 	pstore "github.com/libp2p/go-libp2p-peerstore"
 	ma "github.com/multiformats/go-multiaddr"
-	"google.golang.org/grpc"
 )
 
 // Client defines methods that can be implemented by any type handling p2p communications.
 type Client interface {
-	OpenClient(id identity.Identity) (p2ppb.P2PServiceClient, error)
+
+	// GetSignaturesForDocument gets the signatures for document
 	GetSignaturesForDocument(ctx *header.ContextHeader, identityService identity.Service, doc *coredocumentpb.CoreDocument) error
+
+	// after all signatures are collected the sender sends the document including the signatures
+	SendAnchoredDocument(ctx context.Context, id identity.Identity, in *p2ppb.AnchorDocumentRequest) (*p2ppb.AnchorDocumentResponse, error)
+}
+
+func (s *p2pServer) SendAnchoredDocument(ctx context.Context, id identity.Identity, in *p2ppb.AnchorDocumentRequest) (*p2ppb.AnchorDocumentResponse, error) {
```

Comment:
`Client` is not a struct, it's an interface of ease of use, testing and mocking.

Reply:
its defined in the p2p/server.go file, its an existing struct, it implements the `Client` interface so that we can reuse the same struct with its internal values to support both p2p client and server modes. Which makes a lot of sense in a p2p node IMO. But this is not something I wrote, it was there before.

Reply:
maybe we should call it something other than p2pServer but p2pNode?

Reply:
sure, we could do that.